### PR TITLE
Fix stringop-overflow error when building with gcc 12.x.

### LINKED ISSUE #1577: stringop-overflow error when building with gcc 12.x
I upgraded my laptop to Fedora 36 yesterday and evidently inherited a newer `gcc`:

```
$ gcc --version
gcc (GCC) 12.1.1 20220507 (Red Hat 12.1.1-1)
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

This leads to the following error during compilation:

```
[ 50%] Building C object src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c: In function ‘oaes_encrypt’:
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1276:52: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1276 |                         _block[ _block_size + _j ] = _j + 1;
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1269:25: note: at offset 16 into destination object ‘_block’ of size 16
 1269 |                 uint8_t _block[OAES_BLOCK_SIZE];
      |                         ^~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/crypto/CMakeFiles/cncrypto.dir/build.make:286: src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1475: src/crypto/CMakeFiles/cncrypto.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

I will post a patch after filing this issue.

### PR DESCRIPTION
**Pull request by ianmacd**
Fixes #1577, this time actually fixing the problem, and without mangling the whitespace. Ahem.

### CODE REVIEW

**Comment by jagerman** [comment]
I can't reproduce this warning, using gcc 12.1.0.  What options/flags are you using to build it?

File: src/crypto/oaes_lib.c [hunk 1]
```diff
@@ -1272,8 +1272,10 @@ OAES_RET oaes_encrypt( OAES_CTX * ctx,
 		memcpy( _block, c + 2 * OAES_BLOCK_SIZE + _i, _block_size );
 		
 		// insert pad
-		for( _j = 0; _j < OAES_BLOCK_SIZE - _block_size; _j++ )
-			_block[ _block_size + _j ] = _j + 1;
+		for( _j = 0; _j < OAES_BLOCK_SIZE - _block_size; _j++ ) {
+			uint8_t offset = _block_size + _j;
+			_block[ offset ] = _j + 1;
+		}
```
Comment (jagerman) [inline src/crypto/oaes_lib.c:1278]: Does the warning go away if you change the condition in the original for loop to:

`_block_size + _j < OAES_BLOCK_SIZE`

(Which is arguably clearer code anyway).

Comment (ianmacd) [reply src/crypto/oaes_lib.c:1278]: Sadly not:

```
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c: In function ‘oaes_encrypt’:
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1276:52: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1276 |                         _block[ _block_size + _j ] = _j + 1;
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1269:25: note: at offset 16 into destination object ‘_block’ of size 16
 1269 |                 uint8_t _block[OAES_BLOCK_SIZE];
      |                         ^~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/crypto/CMakeFiles/cncrypto.dir/build.make:286: src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1475: src/crypto/CMakeFiles/cncrypto.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Comment (jagerman) [reply src/crypto/oaes_lib.c:1278]: > The problem is with line 1276, not 1275.

I know, but for some reason gcc isn't noticing that that variable can't be too high to begin with, and was hoping rearranging the condition would make it notice.  But sadly not.

**Comment by jagerman** [comment]
It turns out that 90% of this code (including the warning-inducing line here) isn't used at all, so I've made PR #1581 to rip most of it out instead.

**Pull request closed without merging**